### DISCUSSION TITLE
Testing new functionality for error reporting from Swedeb

### DISCUSSION THREAD

**Discussion by rebeckahw** [swerik-project/the-swedish-parliament-corpus#27] [Report Errors]
Info from Swedeb regarding speech:
{"id":"prot-202223--135_032","gender":"Kvinna","party":"M","year":2022,"speaker":"Maria Malmer Stenergard"}

There is probably a segmentation error for this speech, it contains only the following text:
Tryck: Elanders Sverige AB, Vällingby 2023

**Comment by MansMeg** [comment]
This is excellent. Although, I guess this might be better as an issue? Or what does @BobBorges thinks?

**Reply by BobBorges** [reply]
@MansMeg This is probably a good test case for accepting a correction from outside the Swerik organization. @rebeckahw would you be able to fork the repo, implement a corrections and submit a pull request? Then we have to decide how to track your correction so it doesn't get overwritten.

**Reply by rebeckahw** [reply]
This functionality is meant for users for the Swedeb interface. They should be able to report errors, but are not expected to fix anything. (I am not really sure what the fix for this particular error would be in any case.) @fredrik1984 suggested this solution with adding errors as discussion posts and let Swerik decide if they actually are errors, and what to do with them when that is the case.

**Reply by fredrik1984** [reply]
The idea is to enable users of SweDeb to report a speech with an error in the data (ocr, segmentation, metadata etc) back to Swerik. Since we don't know how much error reporting traffic there will be, we decided that this could be a first easy option for all involved. 

However, @Bob is this error reporting example clear to you? Or should we change something from the SweDeb side? For example, here it would be good to have the speech ID (with a specific link to that speech in the XML or something), right?

**Reply by MansMeg** [reply]
Yes. But I think it is probably better to file this as issues rather than as discussion. 

Also, we should not require swedeb users to do a fix. But we should also not state that they should not. The ideal situation is if swedeb users also fix errors. But if they cant its better they just report it as an issue.

**Reply by MansMeg** [reply]
And yes, I guess the segment id would be helpful to report as well. Ie the position of the error. I dont see why there is a person reported here if it is a segmentation error?

**Reply by fredrik1984** [reply]
Currently, when a SweDeb user "stands in a speech", the user can click on error report, copy speech metadata, then in the discussion thread add more information, as Rebecka did above.

Is it possible to automatically create issues of these error reports from the discussion thread?

**Reply by MansMeg** [reply]
Not that I can see. So I would suggest it to open up a new issue instead with tag ”error in data”.

So the error is essentially connected to a speech here? I would then include the xml:ids that bob mentions of that speech. Thats the way we can map to the original data.

**Reply by MansMeg** [reply]
Btw, Im really happy this is happening.

**Reply by fredrik1984** [reply]
OK. We can discuss how to best create issues of these error reports.

Yes, the error report will essentially always be connected to a specific speech. @rebeckahw can we add xml:ids for each speech too and have that as metadata for each speech?

**Reply by rebeckahw** [reply]
Anything is possible :) 
Looking in the source xml file, it seems like the xml:id is the id minus the order number (_032 in this case), so that would be an easy fix. (More detailed info about where exactly in the source data this segment comes from would probably be a future feature.)

**Reply by MansMeg** [reply]
Great! The id will help a lot. The rest would probably be solved in a discussion with the person reporting the error.

**Reply by fredrik1984** [reply]
Ok, good! Excellent if you @rebeckahw could add that to the speech metadata in SweDeb. We only need it when the user copies the speech metadata to the error report.

**Reply by rebeckahw** [reply]
Sure!

**Comment by BobBorges** [comment]
It would be best to report the file and the `xml:id` attribute of the offending element. We don't (yet) have speech IDs -- these are a sewdeb invention.
